pkg/controllers/data.go

```diff
@@ -179,3 +179,19 @@ func getQueryParams(r *http.Request) (query *entities.Query, err error) {
 		FinishDate: finishDate,
 	}, nil
 }
+func (d *DataController) writeResponseByError(w http.ResponseWriter, err error) {
+	switch err.Error() {
+	case "user is not authorized to list thing's data":
+		d.writeResponse(w, http.StatusBadRequest, err.Error())
+	case "no access token provided":
+		d.writeResponse(w, http.StatusBadRequest, err.Error())
+	case "deviceID not provided":
+		d.writeResponse(w, http.StatusBadRequest, err.Error())
+	case "failed to parse ID from string to int":
+		d.writeResponse(w, http.StatusUnprocessableEntity, err.Error())
+	case "failed to valid token":
```

Comment:
I think we should return an unauthorized error instead of a bad request when the token is invalid.

Reply:
Thank you for the suggestion @netoax. When I was reading the documentation about rest API's, I saw the Unauthorized code, but this code is returned when the authentication is made by the WWW-Authenticate header field. Since our application authenticates the user with the normal header, I don't know if the Unauthorized error is correct. [Here](https://restfulapi.net/http-status-codes/) is the documentation that I used as a parameter.

Reply:
Hm, I think the post is referring to sending a message, in the **response**, to show the user how to proceed if the token is invalid. Besides, the error itself isn't tied with a specific way of interacting with the protected resource, i.e. the `token` could be provided via `auth_token`, `Authorization`, in the payload, doesn't matter in general.

Reply:
Done!